superset-frontend/src/components/FlashProvider/index.tsx

```diff
@@ -37,21 +36,19 @@ const flashObj = {
   success: 'addSuccessToast',
 };
 
-class FlashProvider extends React.PureComponent<Props> {
-  componentDidMount() {
-    const flashMessages = this.props.common.flash_messages;
-    flashMessages.forEach(message => {
+const FlashProvider: React.FC<Props> = ({ children, messages }) => {
```

Comment:
I haven't had any problems with this style of component declaration before, the debugger seems to pick up the names just fine. Are you sure it's an issue?

Reply:
@ktmud 
> 1. Use named function instead of arrow functions, make it easier to find in debugging code
I prefer arrow function because it is more performative and easier to read. But in this case, it's just my taste. I can change this without any problems

> 2. Prefix props with component name, make it less likely to have export conflicts
I put a prefix when the interface is exported. This is not the case for this component/file. If someone in the future puts an export on the interface, they need to change the name too.

Reply:
@suddjian it's probably not a big issue anymore because of sourcemaps, but it used to be useful because function names are captured in error stacks which helps you to locate bad code in case of JS errors. I don't see additional benefits of arrow functions in creating React components so I always follow the named function style.

It's personal preference, but feel free to choose whatever style you like until we adopt a styleguide just for consistency.

@yardz Prefix would be preferred even if the props are not exported, because you never know whether you will export them later or add more components in the same file. Always adding the prefix helps you not to have to worry about this.

Reply:
@ktmud this is only true for anonymous default exports. In the case of `const CoolComponent: React.FC<Props> = props => ...` react is able to add the component name to stack traces based on the variable name, this also allows for the use of the `React.FC` interface which adds the `children` prop.

Reply:
@nytai Indeed, JSX will add `name` to components created by arrow functions. My comments were based on past experience pre-React world.

Re: the automatically added `children` prob. I'm not sure how useful is this. A component either needs or not need the component prob. If it does not handle `children`, then `children` should not be an accepted prop; if it does, then it makes sense to be more explicit.

Reply:
>Whether you will export them later

In this case, the person who places an export simply adds the prefix. It’s not complex to do that, just read rsrs
or add more components in the same file. Always adding the prefix helps you not to have to worry about this.

> add more components in the same file

This is an extremely bad practice. Glad that not adding a prefix gets in the way to add more than one component per file.